hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/migration/owner/SupportTeamToOwnerMigratorSpec.groovy

```diff
@@ -0,0 +1,110 @@
+    def migrator = new SupportTeamToOwnerMigrator(groupService, topicService, subscriptionService)
+
+    def "should count migrated topics and subcriptions"() {
+        given:
+        topicService.listTopics("group") >> [topic("group.topic").withOwner(null).build()]
+        subscriptionService.listSubscriptions(TopicName.fromQualifiedName("group.topic")) >> [
+                subscription(TopicName.fromQualifiedName("group.topic"), "sub").withOwner(null).withSupportTeam("Alpha").build()
+        ]
+
+        when:
+        def stats = migrator.execute("some-source", SupportTeamToOwnerMigrator.OwnerExistsStrategy.SKIP)
+
+        then:
+        stats.topics().migrated() == 1
+        stats.topics().skipped().isEmpty()
+        stats.subscriptions().migrated() == 1
+        stats.subscriptions().skipped().isEmpty()
+    }
+
+    def "should skip topics and subscriptions that already have an owner when using SKIP strategy"() {
+        given:
+        topicService.listTopics("group") >> [topic("group.topic").withOwner(new OwnerId("a", "b")).build()]
+        subscriptionService.listSubscriptions(TopicName.fromQualifiedName("group.topic")) >> [
+                subscription(TopicName.fromQualifiedName("group.topic"), "sub").withOwner(new OwnerId("a", "b")).withSupportTeam("Alpha").build()
+        ]
+
+        when:
+        def stats = migrator.execute("some-source", SupportTeamToOwnerMigrator.OwnerExistsStrategy.SKIP)
+
+        then:
+        stats.topics().migrated() == 0
+        stats.topics().skipped() == ["owner already exists": 1]
+        stats.subscriptions().migrated() == 0
+        stats.subscriptions().skipped() == ["owner already exists": 1]
+    }
+
+    def "should override owners for topics and subscriptions that already have an owner when using OVERRIDE strategy"() {
+        given:
+        topicService.listTopics("group") >> [topic("group.topic").withOwner(new OwnerId("a", "b")).build()]
+        subscriptionService.listSubscriptions(TopicName.fromQualifiedName("group.topic")) >> [
+                subscription(TopicName.fromQualifiedName("group.topic"), "sub").withOwner(new OwnerId("a", "b")).withSupportTeam("Alpha").build()
+        ]
+
+        when:
+        def stats = migrator.execute("some-source", SupportTeamToOwnerMigrator.OwnerExistsStrategy.OVERRIDE)
+
+        then:
+        stats.topics().migrated() == 1
+        stats.topics().skipped().isEmpty()
+        stats.subscriptions().migrated() == 1
+        stats.subscriptions().skipped().isEmpty()
+    }
+
+    def "should skip topics and subscriptions that failed to get updated"() {
+        given:
+        topicService.listTopics("group") >> [
+                topic("group.invalid").withOwner(null).build(),
+                topic("group.error").withOwner(null).build(),
+                topic("group.success").withOwner(null).build()
+        ]
+        subscriptionService.listSubscriptions(TopicName.fromQualifiedName("group.invalid")) >> [
```

Comment:
some constants / shorthand methods would be fit to have these more readable here